File: src/main/scala/com/vc/medium/L452.scala
package com.vc.medium

object L452 {
  def findMinArrowShots(points: Array[Array[Int]]): Int = {
    object BalloonSorting extends Ordering[Array[Int]] {
      def compare(x: Array[Int], y: Array[Int]): Int = x(1).compareTo(y(1))
    }
    if(points.length == 0) return 0
    scala.util.Sorting.quickSort(points)(BalloonSorting)
    var arrowCount = 1
    var arrowPos = points(0)(1)
    var i = 0
    val n = points.length
    while(i < n) {
      if(arrowPos < points(i)(0)) {
        arrowCount += 1
        arrowPos = points(i)(1)
      }
      i += 1
    }
    arrowCount
  }
}

File: src/main/scala/com/vc/medium/L452.java
package com.vc.medium;

import java.util.*;

class L452 {
    public int findMinArrowShots(int[][] points) {
        if(points.length == 0) return 0;
        Arrays.sort(points, (a, b) -> a[1] - b[1]);

        int arrowCount = 0;
        long arrowLocation = Long.MIN_VALUE;
        for(int i = 0; i < points.length; i++) {
            int start = points[i][0];
            int end = points[i][1];
            //System.out.println("start: "+start+" end: "+end+" arrowLocation: "+arrowLocation);
            if(start > arrowLocation) {
                arrowCount++;
                arrowLocation = end;
                //System.out.println(arrowLocation+" "+arrowCount);
            }
        }
        return arrowCount;
    }
}